python/pyspark/sql/catalog.py

```diff
@@ -139,11 +140,40 @@ def listDatabases(self) -> List[Database]:
             jdb = iter.next()
             databases.append(
                 Database(
-                    name=jdb.name(), description=jdb.description(), locationUri=jdb.locationUri()
+                    name=jdb.name(),
+                    catalog=jdb.catalog(),
+                    description=jdb.description(),
+                    locationUri=jdb.locationUri(),
                 )
             )
         return databases
 
+    def getDatabase(self, dbName: str) -> Database:
+        """Get the database with the specified name.
+        This throws an AnalysisException when the database cannot be found.
+
+        .. versionadded:: 3.4.0
+
+        Parameters
+        ----------
+        dbName : str
+             name of the database to check existence.
+
+        Examples
+        --------
+        >>> spark.catalog.getDatabase("default")
+        Database(name='default', catalog=None, description='default database', ...
```

Comment:
I was not aware of this though:

Is the `catalog=None` because of current catalog is not set? I am asking because I thought every DB should have a catalog.

Reply:
because of the old code path, which intilize a `Database` with `null` catalog:

https://github.com/apache/spark/pull/36968/files#diff-d7036fd31ee9734405f0e877b79e36675d567b6f9acb844034383463e80a54ccR65-R67